src/services/school/hooks/index.js

```diff
@@ -220,6 +220,40 @@ const validateSchoolNumber = (context) => {
 	return context;
 };
 
+// school County
+const validateCounty = async (context) => {
+	if (context && context.data && context.data.county) {
+		const schools = await context.app.service('schools').find({
+			query: {
+				_id: context.id,
+				$populate: 'federalState',
+				$limit: 1,
+			},
+		});
+		const currentSchool = schools.data[0];
+		if (!currentSchool) {
+			throw new Error(`Internal error`);
+		}
+
+		const { county } = context.data;
+		if (
+			!currentSchool.federalState.counties.length ||
+			!currentSchool.federalState.counties.some((c) => JSON.stringify(c._id) === JSON.stringify(county))
```

Comment:
Please let us discuss again how it should use i future. Adding it to federallState make sense if we use it in context of federalState. But if we want to make a lookup to a specific country it make no sense to grep all federalStates. If we want to link the embedded ids of countrys i am note sure if it work with ref.

Reply:
we need to make a lookup when querying the lern-store. The County is taken as paramter in search.